dash-pipeline/bmv2/stages/routing_action_apply.p4

```diff
@@ -15,7 +16,9 @@ control routing_action_apply(
         // because it requires the transforms on the inner packet to be finished in order to
         // get the correct inner packet size and other informations.
         do_action_static_encap.apply(hdr, meta);
+
+        tunnel_stage.apply(hdr, meta);
```

Comment:
this should happen before action apply stage.

Reply:
It does not makes sense. Tunnel stage works on outer layers. If I move it before static_encap, it will mess up the tunnel_pointer.

Reply:
yea, sorry, I probably didn't make it clear. 

In my mind, the "tunnel stage" is the tunnel id to tunnel lookup, just like the routing stage and mapping stage, so the lookup needs to happen before the static encap.

However, like you mentioned, applying tunnel encap for changing the packet should happen after the static encap, and this keeps the tunnel pointer working.

Reply:
Fixed, split lookup from the encap itself.